### PR TITLE
Getting hit by the tram will now smash people through lattice and unplated flooring

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
A common administrative issue and general gripe about Tramstation is the practice of players using lattice or floor tiles to cover up the big open pits people fall down when they're hit by the tram, making it so they're doomed to get hit again and again and again, racking up massive damage and quickly becoming lethal. This PR changes the tram to cause people it hits to smash right through lattice or unplated flooring, suffering some additional damage along the way. 

[](https://i.imgur.com/USQgQzT.png)

If you still want to see people get memed to hell, you can simply take the extra time to add plating to the floor tiles, but at least then admins know that you're purposely trying to get people killed and not just a naïve person thinking that keeping people from falling to the z-level below improves safety rather than massively decreases it.

[](https://i.imgur.com/3ZOXZDs.png)
Support from the resident tram expert, @tralezab 

## Why It's Good For The Game
Hopefully reigns in the tram's extreme lethality when players modify the environment in certain ways

## Changelog

:cl: Ryll/Shaps
balance: Getting hit by the tram and landing on lattice or unplated flooring (on the upper z-level) will now cause you to smash through it and fall to the z-level below at the cost of taking some extra damage. Note that floors with plating will still hold steady, so you can still use flooring to make getting hit by the tram insanely lethal, it just takes a bit more effort.
/:cl:

### CODE REVIEW

**Comment by Mooshimi** [comment]
Your branch and merge names are funny, also I support this

**Comment by Mothblocks** [comment]
image

u gotta fix ur snippie

**Review by Mothblocks — APPROVED** [review]
other than that good

**Comment by SmArtKar** [comment]
I think that this won't fix the issue, people will just additionaly put tiles which will take them just a bit more time.

**Comment by iwishforducks** [comment]
> I think that this won't fix the issue, people will just additionaly put tiles which will take them just a bit more time.

From the PR body:
> but at least then admins know that you're purposely trying to get people killed

**Comment by MMMiracles** [comment]
i was against this until i saw you can still make the tram tunnel hyper-lethal with floor tiles, so this is a fine change

also that branch name lmao

**Comment by Mothblocks** [comment]
To clarify, the problem Shaps is describing is something I've echoed with them for a long time.

People will frequently place tiles on the Tram tracks. This makes the tram super-deadly-no-coming-back-without-a-competent-and-bored-medbay deadly. Unlike something like people blowing it up with field generators, there is a good faith argument to be made that "oh well I just wanted it to be easier to cross the tracks". Whether they're lying or not, it's come up so often (more than half of all Tram rounds I've been in) that it becomes exhausting and I begged for something like this instead.

We are not (just) trying to stop griefers, we're making sure those bad faith griefers lose their good faith argument.

**Comment by Mooshimi** [comment]
> Go go gadget floorbot
> B)

except they die whenever the tram goes by lmfao

**Comment by maxymax13** [comment]
the better solution is make it so trying to place stuff in the tiles the tram goes over instantly gibs and round removes you, gives you a bad note and job bans you from engineering permanently

**Comment by Mooshimi** [comment]
> the better solution is make it so trying to place stuff in the tiles the tram goes over instantly gibs and round removes you, gives you a bad note and job bans you from engineering permanently

it's not engineers who do this most the time from what I've seen. anyone does it including borgs

**Comment by Mothblocks** [comment]
@Ryll-Ryll Ready?

**Comment by Ryll-Ryll** [comment]
ya sure

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -2153,3 +2153,28 @@
...
+		apply_damage(rand(5,20), BRUTE, BODY_ZONE_CHEST)
+		smashed_plating.ScrapeAway(1, CHANGETURF_INHERIT_AIR)
+
+	for(var/atom/iter_atom as anything in loc)
+		if(istype(iter_atom, /obj/structure/lattice))
+			visible_message(span_danger("[src] is thrown violently into [iter_atom], smashing through it and punching straight through!"),
+				span_userdanger("You're thrown violently into [iter_atom], smashing through it and punching straight through!"))
+			apply_damage(rand(5,10), BRUTE, BODY_ZONE_CHEST)
+			var/obj/structure/bye_bye = iter_atom
+			bye_bye.deconstruct(FALSE)
```
Comment (Mothblocks) [inline code/modules/mob/living/living.dm:2180]:
```suggestion
	for(var/obj/structure/lattice/lattice in loc)
		visible_message(span_danger("[src] is thrown violently into [lattice], smashing through it and punching straight through!"),
			span_userdanger("You're thrown violently into [lattice], smashing through it and punching straight through!"))
		apply_damage(rand(5,10), BRUTE, BODY_ZONE_CHEST)
		lattice.deconstruct(FALSE)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by LemonInTheDark** [comment]
@Mothblocks I want this. Any reason not to go?

**Comment by Mothblocks** [comment]
i was waiting for ci

**Pull request merged**